### PR TITLE
Fix sign-in issue due to calling getToken in main thread

### PR DESCRIPTION
**Pull request by zhiyua-git**
Related issue: https://github.com/aws-amplify/aws-sdk-android/issues/956

*Issue #, if available:*
956

*Description of changes:*
Move isSignedIn() to AsyncTask to avoid NetworkOnMainThreadException and crash.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/StartUpActivity.java [hunk 1]
```diff
@@ -19,31 +20,37 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                            .build(),
-                    new Callback<UserStateDetails>() {
-                        @Override
-                        public void onResult(UserStateDetails result) {
-                            Log.d(TAG, "onResult: User signed-in " + result.getUserState());
-                        }
+        AsyncTask.execute(new Runnable() {
+            @Override
+            public void run() {
+                if (auth.isSignedIn()) {
```
Comment (kvasukib) [inline AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/StartUpActivity.java]: Is it possible to just call `isSignedIn()` from a background thread or AsyncTask and wait until the call returns and then proceed to start activity or showSignIn based on the response?

Comment (zhiyua-git) [reply AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/StartUpActivity.java]: I would rather expect AWS mobile client will do this Async task and return response inside the isSignedIn().

**Comment by gaganAriel** [comment]
Please let me know run time how i can change folders which are inside a bucket

**Comment by palpatim** [comment]
@gaganAriel

> Please let me know run time how i can change folders which are inside a bucket

Please open an issue on the issues page with your question, and add more detail to help us answer.

File: AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/StartUpActivity.java (hunk 1 above)
Comment (faisalvv) [reply AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/StartUpActivity.java]: auth.getTokens(new Callback<Tokens>() {
                        @Override
                        public void onResult(Tokens result) {
                            ActivityUtils.startActivity(thisActivity, SimpleNavActivity.class);
                        }

                        @Override
                        public void onError(Exception e) {

                        }
                    });

**Comment by raphkim** [comment]
Closing in favor of #332

**Pull request closed without merging**